### PR TITLE
Validate VkBindImagePlaneMemoryInfo is in pNext when binding planes

### PR DESCRIPTION
**Pull request by aitor-lunarg**
I couldn't find any VUID (I think there's none and one should be added) that states a `VkBindImagePlaneMemoryInfo` is needed in the `pNext` of `VkBindImageMemoryInfo` when binding planes of disjoint images.

Without adding it to `pNext`, there's no way to specify to which plane we are binding the memory, and the spec states the following `In order to bind planes of a disjoint image, add a VkBindImagePlaneMemoryInfo structure to the pNext chain of VkBindImageMemoryInfo (https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkBindImageMemoryInfo.html).`.

### CODE REVIEW

**Comment by aitor-lunarg** [comment]
#4122 also has the same modification to the test, but I think having it in a different PR with a new VUID to explain should be a better approach than just modifying the test.

**Review by Tony-LunarG — APPROVED** [review]
LGTM but could use a test for UNASSIGNED-VkBindImageMemoryInfo-pNext-missing-VkBindImagePlaneMemoryInfo

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -15312,6 +15312,13 @@ bool CoreChecks::ValidateBindImageMemory(uint32_t bindInfoCount, const VkBindIma
             const auto plane_info = LvlFindInChain<VkBindImagePlaneMemoryInfo>(bind_info.pNext);
             auto mem_info = Get<DEVICE_MEMORY_STATE>(bind_info.memory);
 
+            if (image_state->disjoint && plane_info == nullptr) {
+                skip |= LogError(bind_info.image, "UNASSIGNED-VkBindImageMemoryInfo-pNext-missing-VkBindImagePlaneMemoryInfo",
```
Comment (ncesario-lunarg) [inline layers/core_validation.cpp:15316]: `UNASSIGNED-VkBindImageMemoryInfo-pNext-missing-VkBindImagePlaneMemoryInfo` should be set in `core_validation_error_enums.h` and then the constant used here. Then as Tony suggests, there should be a test for this.

**Review by ncesario-lunarg — CHANGES_REQUESTED** [review]
LGTM with a minor change + Tony's suggestion.

**Review by ziga-lunarg — APPROVED** [review]
LGTM other than what Tony and Nate pointer out

File: layers/core_validation.cpp (hunk 1 above)
Comment (aitor-lunarg) [reply layers/core_validation.cpp:15316]: Thank you for pointing it out! I was not aware of this procedure.

**Review by Tony-LunarG — APPROVED** [review]
LGTM

File: layers/core_validation.cpp (hunk 1 above)
Comment (ncesario-lunarg) [reply layers/core_validation.cpp:15316]: Yeah, this information should probably be mentioned (or at least reachable) via CONRIBUTING.md. I've added a TODO to add this.

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 2]
```diff
@@ -1665,6 +1665,8 @@ TEST_F(VkLayerTest, BindInvalidMemoryYcbcr) {
 
         // Bind disjoint with BindImageMemory instead of BindImageMemory2
         m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-vkBindImageMemory-image-01608");
+        m_errorMonitor->SetDesiredFailureMsg(kErrorBit,
+                                             "UNASSIGNED-VkBindImageMemoryInfo-pNext-missing-VkBindImagePlaneMemoryInfo");
```
Comment (ncesario-lunarg) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:1669]: We don't seem to be consistent on this throughout the tests, but I would think we'd want to use the constant here (and below). Up to you on what makes more sense.

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

**Review by jeremyg-lunarg — APPROVED** [review]

**Pull request merged**